subworkflows/local/utils_nfcore_mcmicro_pipeline/main.nf

```diff
@@ -180,6 +180,9 @@ def validateInputParameters() {
         error "You can only provide a cellpose model if you have selected cellpose as one of your segmentation methods"
     }
 
+    if (params.tma_dearray && !params.pixel_size) {
+        error "You must provide an pixel_size in microns when the sample is a tissue microarray (TMA)."
```

Comment:
Could we add a test to `tests/initialisation.nf.test` for this? I played around a bit and came up with the following. Straighforward setup, but it took me some playing around to get the stdout check working as intended. I had to dig into the nf-test code to figure out that `workflow.errorReport` (which is what I thought I should be checking at first) comes via the Nextflow variable of the same name which contains "Detailed error of the task that caused the workflow execution to fail" but is not set when the pipeline itself calls `error()`. Also I would have expected `error` to write to stderr, but it went to stdout.

```groovy
        test("Error on tma_dearray without pixel_size") {
            when {
                params {
                    input_cycle = "${projectDir}/assets/samplesheet-test.csv"
                    marker_sheet = "${projectDir}/assets/markers_1_sp.csv"
                    tma_dearray = true

                }
                workflow {
                    """
                    input = [false, false, false, false, [], '$outputDir', params.input_cycle, [], params.marker_sheet]
                    """
                }
            }
            then {
                assert workflow.failed
                assert workflow.stdout*.contains("pixel_size").any()
            }
        },
```

Reply:
Yes, that's a great idea! This is the perfect place for a negative test.  Yeah, that's odd that it's stdout instead of stderr.
I'll try this.

Reply:
works great! Thanks!